boto/s3/key.py

```diff
@@ -1549,7 +1553,7 @@ def _get_file_internal(self, fp, headers=None, cb=None, num_cb=10,
             cb(data_len, cb_size)
         try:
             for bytes in self:
-                fp.write(bytes)
+                boto.utils.ttyprint(bytes, file=fp, end='')
```

Comment:
The name on this method seems to be misleading -- we're not necessarily writing to a tty, correct? `fp` could refer to a file descriptor that isn't a terminal.  Maybe the method should be named `write_to_file_or_tty`?

Reply:
Sounds good to me. What about `write_to_fd` or `write_to_file_descriptor`? A smidge shorter for the former, and file descriptor covers both, especially since `file=` should be able to take any kind of fd.

Reply:
Sounds good. Might even be worth revisiting the logic to see if we can break it up into something like:

```
if fd_is_tty():
  write_to_tty(data)
else:  # fd is a file
  write_to_file(data)
```

... if that's possible.  It's a bit tough to follow as-is.

Reply:
Yeah, it's due for a refactor, I'll do break it up a bit :+1:

Reply:
Since this is the single and only call in boto to ttyprint / ttywrite, I'm going to axe those functions from boto entirely and just do six.ensure_binary() here. The refactor to those functions I'll include in gsutil where those functions are used more widely